server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -376,6 +377,8 @@ public abstract class BlobStoreRepository extends AbstractLifecycleComponent imp
      */
     private final int maxSnapshotCount;
 
+    private ShardSnapshotWorkers shardSnapshotWorkerPool;
```

Comment:
nit: I'd prefer to name this identical to the type:

```suggestion
    private ShardSnapshotWorkers shardSnapshotWorkers;
```